service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java

```diff
@@ -99,64 +96,166 @@ public void setRestTemplate(RestTemplate restTemplate) {
+                            RemremGenerateServiceConstants.JSON_ERROR_MESSAGE_FIELD,
+                            status, e.getMessage(), "fail", errorResponse, status.value());
+                }
             }
-            else {
-                return new ResponseEntity<>(parser.parse(e1.getMessage()), HttpStatus.UNPROCESSABLE_ENTITY);
+            return createResponseEntity(RemremGenerateServiceConstants.JSON_STATUS_CODE, RemremGenerateServiceConstants.JSON_STATUS_RESULT,
+                    RemremGenerateServiceConstants.JSON_ERROR_MESSAGE_FIELD,
+                    HttpStatus.BAD_REQUEST, e.getMessage(), "fail", errorResponse, HttpStatus.BAD_REQUEST.value());
+        } else if (e instanceof JsonSyntaxException) {
+            log.error("Failed to parse JSON: ", exceptionMessage);
+            return createResponseEntity(RemremGenerateServiceConstants.JSON_STATUS_CODE, RemremGenerateServiceConstants.JSON_STATUS_RESULT, RemremGenerateServiceConstants.JSON_ERROR_MESSAGE_FIELD,
+                    HttpStatus.BAD_REQUEST, e.getMessage(), "fail", errorResponse, HttpStatus.BAD_REQUEST.value());
+        } else {
+            log.error("Unexpected exception caught", exceptionMessage);
+            return createResponseEntity(RemremGenerateServiceConstants.JSON_STATUS_CODE, RemremGenerateServiceConstants.JSON_STATUS_RESULT,
+                    RemremGenerateServiceConstants.JSON_ERROR_MESSAGE_FIELD,
+                    HttpStatus.INTERNAL_SERVER_ERROR, exceptionMessage, "fail", errorResponse, HttpStatus.INTERNAL_SERVER_ERROR.value());
+        }
+    }
+
+    /**
+     * This helper method basically generate or process one event
+     * @return JsonObject generated event
+     */
+
+    public JsonObject processEvent(String msgProtocol, String msgType, Boolean failIfMultipleFound,
+                                   Boolean failIfNoneFound, Boolean lookupInExternalERs, int lookupLimit,
+                                   Boolean okToLeaveOutInvalidOptionalFields, JsonObject jsonObject) throws REMGenerateException, JsonSyntaxException {
+        JsonObject eventResponse = new JsonObject();
+        JsonElement parsedResponse;
+
+        JsonObject event = erLookup(jsonObject, failIfMultipleFound, failIfNoneFound, lookupInExternalERs, lookupLimit);
+        MsgService msgService = getMessageService(msgProtocol);
+
+        if (msgService != null) {
+            String response = msgService.generateMsg(msgType, event, isLenientEnabled(okToLeaveOutInvalidOptionalFields));
+            parsedResponse = parser.parse(response);
+
+            JsonObject parsedJson = parsedResponse.getAsJsonObject();
+
+            if (!parsedJson.has(RemremGenerateServiceConstants.JSON_ERROR_MESSAGE_FIELD)) {
+                return parsedJson;
+            } else {
+                eventResponse.addProperty(RemremGenerateServiceConstants.JSON_STATUS_CODE, HttpStatus.BAD_REQUEST.value());
```

Comment:
Use createResponse() method. Please, check if the method can be used at other places.